examples/esn.py

```diff
@@ -17,79 +23,85 @@
+conn = iodata.load_file('connectivity.npy')
 
 # select one subject
 subj_id = 0
-conn = conn[:,:,subj_id]
+conn = conn[:, :, subj_id]
 n_reservoir_nodes = len(conn)
 
 # scale conenctivity weights between [0,1]
 conn = (conn - conn.min())/(conn.max() - conn.min())
 
 # normalize connectivity matrix by the spectral radius.
-from scipy.linalg import eigh
 
 ew, _ = eigh(conn)
-conn  = conn / np.max(ew)
+conn = conn / np.max(ew)
 
 ###############################################################################
 # Second let's get the data to perform the task. We first generate the data and
 # then we split it into training and test sets. 'x' corresponds to the input
 # signals and 'y' corresponds to the output labels.
-from conn2res import iodata
 
 # get trial-based dataset for task
-task = 'PerceptualDecisionMaking' 
+task = 'PerceptualDecisionMaking'
 n_trials = 1000
 x, y = iodata.fetch_dataset(task, n_trials=n_trials)
 
 # visualize task data
 iodata.visualize_data(task, x, y, plot=True)
 
+# length of sequence within each trial
+seq_len = x[0].shape[0]
+
+# get sample weights of data which reflect the decision time points
+sample_weight = [np.hstack((np.zeros(seq_len-1), np.ones(1)))
+                 for i in range(n_trials)]
+
 # split trials into training and test sets
-x_train, x_test = iodata.split_dataset(x)
-y_train, y_test = iodata.split_dataset(y)
+x_train, x_test, y_train, y_test = iodata.split_dataset(x, y, axis=0)
+sample_weight_train, sample_weight_test = iodata.split_dataset(
+    sample_weight, axis=1)
 
 ###############################################################################
 # Third we will simulate the dynamics of the reservoir using the previously
 # generated input signal x (x_train and x_test).
 
 # define set of input nodes
-ctx  = np.load(os.path.join(DATA_DIR, 'cortical.npy'))
-subctx_nodes = np.where(ctx == 0)[0] # we use subcortical regions as input nodes
+ctx = iodata.load_file('cortical.npy')
+# we use subcortical regions as input nodes
+subctx_nodes = np.where(ctx == 0)[0]
 
 n_features = x_train.shape[1]
-input_nodes  = np.random.choice(subctx_nodes, n_features) # we select a randon set of input nodes
-output_nodes = np.where(ctx == 1)[0] # we use cortical regions as output nodes
+# we select a randon set of input nodes
+input_nodes = np.random.choice(subctx_nodes, n_features)
+output_nodes = np.where(ctx == 1)[0]  # we use cortical regions as output nodes
 
 # create input connectivity matrix, which defines the connec-
 # tions between the input layer (source nodes where the input signal is
 # coming from) and the input nodes of the reservoir.
 w_in = np.zeros((n_features, n_reservoir_nodes))
-w_in[np.ix_(np.arange(n_features), input_nodes)] = 10.0 # factor that modulates the activation state of the reservoir
+# factor that modulates the activation state of the reservoir
+w_in[np.ix_(np.arange(n_features), input_nodes)] = 10.0 * np.eye(n_features)
```

Comment:
This fix allows to calculate the difference between two input signals which is essential, for instance, in the PerceptualDecisionMaking task. Before the fix, the input signals were summed up when calculating that state of the input nodes so the RC no longer has access to the input signals separately to be able to perform the task.